R/fcast.R

```diff
@@ -220,26 +216,6 @@ dcast.data.table <- function(data, formula, fun.aggregate = NULL, sep = "_", ...
       # removed 'setcolorder()' here, #1153
     setattr(ans, 'names', c(lhsnames, allcols))
     setDT(ans); setattr(ans, 'sorted', lhsnames)
-  } else {
-    # formula is of the form x + y ~ . (rare case)
```

Comment:
Added a comment for Arun on the original PR here:

https://github.com/Rdatatable/data.table/commit/25a74dfd8b738e5e5556144d38713b6b003a56f2

I don't think this branch is possible specifically due to L116:

```
lvars = lapply(lvars, function(x) if (length(x)) x else quote(`.`))
```